controllers/remote/cluster.go

```diff
@@ -18,53 +18,41 @@ package remote
 
 import (
 	"github.com/pkg/errors"
-	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 	restclient "k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
 	kcfg "sigs.k8s.io/cluster-api/util/kubeconfig"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-// ClusterClient is an interface encapsulating methods
-// to access a remote cluster.
-type ClusterClient interface {
-	RESTConfig() *restclient.Config
-	CoreV1() (corev1.CoreV1Interface, error)
-}
-
-// clusterClient is a helper struct to connect to remote workload clusters.
-type clusterClient struct {
-	restConfig *restclient.Config
-	cluster    *clusterv1.Cluster
-}
+// NewClusterClient returns a Client for interacting with a remote Cluster using the given scheme for encoding and decoding objects.
+func NewClusterClient(c client.Client, cluster *clusterv1.Cluster, scheme *runtime.Scheme) (client.Client, error) {
 
-// NewClusterClient creates a new ClusterClient.
-func NewClusterClient(c client.Client, cluster *clusterv1.Cluster) (ClusterClient, error) {
-	kubeconfig, err := kcfg.FromSecret(c, cluster)
+	restConfig, err := RESTConfig(c, cluster)
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to retrieve kubeconfig secret for Cluster %q in namespace %q",
-			cluster.Name, cluster.Namespace)
+		return nil, err
 	}
 
-	restConfig, err := clientcmd.RESTConfigFromKubeConfig(kubeconfig)
+	ret, err := client.New(restConfig, client.Options{Scheme: scheme})
```

Comment:
@vincepri do we need to pass in mapper = `apiutil.NewDynamicRESTMapper(restConfig)`?

Reply:
We should, yes. It seems the plain client still uses the non-dynamic REST mapper